### PR TITLE
feat(@cubejs-client/core): allow making HTTP POST requests to Cube.js API

### PR DESCRIPTION
**Pull request by mnifakram**
**Check List**
- [x] Tests has been run in packages where changes made if available
- [x] Linter has been run for changed code
- [x] Tests for the changes have been added if not covered yet
- [x] Docs have been added / updated if required

**Issue Reference this PR resolves**

 #1558

### CODE REVIEW

**Comment by mnifakram** [comment]
@hassankhan @ovr can you please guys review? Thanks.

File: packages/cubejs-client-core/src/HttpTransport.js [hunk 1]
```diff
@@ -2,34 +2,40 @@ import fetch from 'cross-fetch';
 import 'url-search-params-polyfill';
 
 class HttpTransport {
-  constructor({ authorization, apiUrl, headers = {}, credentials }) {
+  constructor({ authorization, apiUrl, method = 'GET', headers = {}, credentials }) {
```
Comment (hassankhan) [inline packages/cubejs-client-core/src/HttpTransport.js:5]: @ovr Should we also be changing the default to `POST` too?

Comment (vasilev-alex) [reply packages/cubejs-client-core/src/HttpTransport.js:5]: We should prefer the `GET` method whenever possible

**Comment by vasilev-alex** [comment]
So, currently, I can see one limitation. It's not possible to change the `method` once we've created the `cubejsApi` instance.

```js
const transport = new HttpTransport({
  // ...
  method: 'POST'
});

const cubejsApi = cubejs(CUBEJS_TOKEN, {
  apiUrl: `${API_URL}/cubejs-api/v1`,
  transport
});

// Can't change the `method` during runtime
// I want to send this one using `GET`
cubejsApi.load(query);
// and this one using `POST`
cubejsApi.load(query);
```

What if we would pass the HTTP `method` to the load?

```js
cubejsApi.load(query, { method: 'GET' });
cubejsApi.load(query, { method: 'POST' });
```

Or, maybe, we could allow users to override the method by passing it to the `load` method? Do we even need it?

@ovr @hassankhan @mnifakram what do you think?

**Comment by mnifakram** [comment]
@vasilev-alex I don't think that users will switch between `GET` or `POST`, that's an edge case. In real world as a user if I didn't encounter the large URL problem and I only have `POST`, I will not even complain about not having `GET` as an option. 

I don't see any advantages of using `GET` over `POST`. Not to mention that, it is way easier to inspect the query in the request body vs trying to decoded from the request URL.

File: packages/cubejs-client-core/src/HttpTransport.js (hunk 1 above)
Comment (ovr) [reply packages/cubejs-client-core/src/HttpTransport.js:5]: 100% agree with @vasilev-alex, it's better to use GET whenever possible.

**Comment by ovr** [comment]
> I don't see any advantages of using GET over POST.

`GET` requests support cache inside browser, and possible Cube.js will return specific headers to support it.

File: packages/cubejs-client-core/src/HttpTransport.js (hunk 1 above)
Comment (ovr) [reply packages/cubejs-client-core/src/HttpTransport.js:5]: @vasilev-alex @hassankhan Maybe it's better to set up it to `undefined` by default and detects automatically. 

```
method = method || (url.length < 2000 ? 'GET' : 'POST')
```

**Comment by mnifakram** [comment]
Is there anything that I can help with to get this merged? I'm really in need to get this released asap. Thanks

**Comment by himessa** [comment]
Any update on this? I need the ability to handle large queries as well.

File: packages/cubejs-client-core/src/HttpTransport.js (hunk 1 above)
Comment (vasilev-alex) [reply packages/cubejs-client-core/src/HttpTransport.js:5]: @ovr yep, let's do like this. @mnifakram can you please introduce this check?

Comment (mnifakram) [reply packages/cubejs-client-core/src/HttpTransport.js:5]: yes sure thing

Comment (mnifakram) [reply packages/cubejs-client-core/src/HttpTransport.js]: @ovr @vasilev-alex I've just updated it. Please kindly check it. thanks

File: packages/cubejs-client-core/src/index.js [hunk 2]
```diff
@@ -25,11 +25,13 @@ class CubejsApi {
     options = options || {};
     this.apiToken = apiToken;
     this.apiUrl = options.apiUrl || API_URL;
+    this.method = options.method || 'GET';
```
Comment (vasilev-alex) [inline packages/cubejs-client-core/src/index.js:28]: In this case, it will always be `GET` by default

Comment (mnifakram) [reply packages/cubejs-client-core/src/index.js]: oops forgot to change this file, I've updated it again. Thanks

**Pull request merged**